src/main/java/htsjdk/variant/variantcontext/writer/BCF2Writer.java

```diff
@@ -122,6 +122,12 @@
     private VCFHeader lastVCFHeaderOfUnparsedGenotypes = null;
     private boolean canPassOnUnparsedGenotypeDataForLastVCFHeader = false;
 
+    // is the header written to the output stream?
+    private boolean isHeaderWritten;
+
+    // is at least one part of body written to the output stream?
+    private boolean isBodyWritten;
+
```

Comment:
@amilamanoj Is there a reason that the writers need to keep two state variables to track what is essentially a single binary state (the header is frozen) ?